### PR TITLE
fetch_related allowing partial fetches

### PR DESCRIPTION
**Pull request by wojcikstefan**
i.e. where we only fetch *some* of the fields for a related object.

### CODE REVIEW

**Comment by wojcikstefan** [comment]
@thomasst @philfreo review please

File: flask_common/documents.py [hunk 1]
```diff
@@ -198,82 +221,108 @@ def setattr_unchanged(obj, key, val):
             document_class = field.field.document_type
         else:
             raise NotImplementedError('%s class not supported for fetch_related' % field.__class__.__name__)
-        field_cache[field_name] = (field, db_field, document_class)
+        fields_to_fetch = sub_field_dict if isinstance(sub_field_dict, list) else None
```
Comment (thomasst) [inline flask_common/documents.py]: It would be nice to be able to pass in a tuple as well.

File: flask_common/documents.py [hunk 2]
```diff
@@ -198,82 +221,108 @@ def setattr_unchanged(obj, key, val):
...
+    for document_class, fetch_opts in fetch_map.iteritems():
+        qs = document_class.objects.filter(pk__in=fetch_opts['ids']).clear_initial_query()
+
+        # only fetch the requested fields
+        if fetch_opts['fields_to_fetch']:
+            qs = qs.only(*fetch_opts['fields_to_fetch'])
+
+        # since ids are unique, we can set a limit equal to the length
+        # of the ids set (helps with performance)
+        qs = qs.limit(len(fetch_opts['ids']))
```
Comment (thomasst) [inline flask_common/documents.py]: Why does this help performance?

Comment (wojcikstefan) [inline flask_common/documents.py]: I intuitively thought it could help, but that's not true for a PK lookup. Will remove.

**Pull request merged**